driver/csiplugin/controllerserver.go

```diff
@@ -1923,6 +1942,11 @@ func (cs *ScaleControllerServer) CreateSnapshot(ctx context.Context, req *csi.Cr
 				glog.Errorf("snapshot [%s] - Unable to create snapshot. Error [%v]", snapName, snaperr)
 				return nil, status.Error(codes.Internal, fmt.Sprintf("unable to create snapshot [%s]. Error [%v]", snapName, snaperr))
 			}
+			if volumeIDMembers.StorageClassType == STORAGECLASS_ADVANCED {
+				snapIdentifier := fmt.Sprintf("%s:%s", filesystemName, filesetName)
+				snapReadyDetails := CGSnapReadyDetails{true, snapName}
```

Comment:
what if we delete the entry from map at this point instead of setting it to true ?